src/pretix/static/pretixpresale/js/widget/widget.js

```diff
@@ -256,7 +256,7 @@ Vue.component('availbox', {
             }
         },
         unavailability_reason_message: function () {
-            var reason = this.item.has_variations ? this.variation.current_unavailability_reason : this.item.current_unavailability_reason;
+            var reason = this.item.current_unavailability_reason || this.variation?.current_unavailability_reason;
```

Comment:
I am not sure about our browser support for the widget, but I think we try to keep it as backwards compatible as possible. So IMHO we should avoid `?.` notation here. I am not sure, why that change is needed here – or can unavailability_reason be set on the containing item even if it has variations? I assumed `unavailability_reason` is always set on each variation, if an item has variations. I might be wrong though.

Reply:
Sorry, I have not read you description correctly, that the item is not available, but variations are. So your code is right, but I still think we should avoid `?.` for backwards-compatibility.

```suggestion
            var reason = this.item.current_unavailability_reason || (this.variation ? this.variation.current_unavailability_reason : '');
```

Reply:
Yes, this is a recent feature ([2020](https://caniuse.com/?search=%3F.%20)), but, we already use this anyway (I had checked it with ctrl-F for `?.` on this file, which has has three other matches), so no new dependency introduced.

Reply:
Oh, you are right. Sorry, if nobody complained before, then we can leave it like that.